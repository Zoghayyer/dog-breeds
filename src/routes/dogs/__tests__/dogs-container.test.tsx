import React from 'react';
import { shallow } from 'enzyme';
import {
  DispatchProps,
  DogsContainer,
  StateProps,
} from '../../dogs/dogs-container';

describe('(Container) DogsContainer', () => {
  let component;
  const defaultProps: DispatchProps & StateProps = {
    dogsBreedDogs: [],
    dogsBreedRequestGetIsLoading: false,
    dogsBreedsAllGetIsLoading: false,
    dogsBreedsAllRequestGet: () => ({}),
    favoriteDogs: [],
    showLoadMore: false,
  };

  const shallowDogsContainer = (extendedProps: Partial<DispatchProps & StateProps> = {}) => {
    component = shallow(<DogsContainer {...defaultProps} {...extendedProps} />);
  };

  beforeEach(() => shallowDogsContainer());

  it('renders dogs view component if dogsBreedsAllGetIsLoading is false', () => {
    shallowDogsContainer({ dogsBreedsAllGetIsLoading: false });
    expect(component).toMatchSnapshot();
  });

  it('renders the loader component if dogsBreedsAllGetIsLoading is true', () => {
    shallowDogsContainer({ dogsBreedsAllGetIsLoading: true });
    expect(component).toMatchSnapshot();
  });

  describe('(Method) componentDidMount', () => {
    it('calls dogsBreedsAllRequestGet', () => {
      const props = {
        ...defaultProps,
        dogsBreedsAllRequestGet: jest.fn().mockResolvedValue({}),
      };
      component = new DogsContainer(props);
      return component.componentDidMount()
        .then(() => {
          expect(props.dogsBreedsAllRequestGet).toHaveBeenCalled();
        });
    });
  });

  describe('(Method) showLoadMore', () => {
    it('returns false if showLoadMore is falsy', () => {
      const props = {
        ...defaultProps,
        dogsBreedRequestGetIsLoading: false,
        showLoadMore: false,
      };
      component = new DogsContainer(props);
      expect(component.showLoadMore()).toBeFalsy();
    });

    it('returns false if dogsBreedRequestGetIsLoading is truthy', () => {
      const props = {
        ...defaultProps,
        dogsBreedRequestGetIsLoading: true,
        showLoadMore: true,
      };
      component = new DogsContainer(props);
      expect(component.showLoadMore()).toBeFalsy();
    });

    it('returns true if showLoadMore is truthy and dogsBreedRequestGetIsLoading is falsy', () => {
      const props = {
        ...defaultProps,
        dogsBreedRequestGetIsLoading: false,
        showLoadMore: true,
      };
      component = new DogsContainer(props);
      expect(component.showLoadMore()).toBeTruthy();
    });
  });
});
