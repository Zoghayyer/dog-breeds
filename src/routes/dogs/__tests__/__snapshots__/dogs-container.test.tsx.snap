// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) DogsContainer renders dogs view component if dogsBreedsAllGetIsLoading is false 1`] = `
<DogsView
  dogsBreedDogs={Array []}
  favoriteDogs={Array []}
  isLoading={false}
  showLoadMore={false}
/>
`;

exports[`(Container) DogsContainer renders the loader component if dogsBreedsAllGetIsLoading is true 1`] = `
<Loader
  className="loader-container"
  size="sm"
/>
`;
